shell/platform/darwin/common/framework/Source/FlutterStandardCodecHelper.c

```diff
@@ -0,0 +1,189 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/common/framework/Source/FlutterStandardCodecHelper.h"
+#include <stdint.h>
+
+void FlutterStandardCodecHelperReadAlignment(unsigned long* location,
+                                             uint8_t alignment) {
+  uint8_t mod = *location % alignment;
+  if (mod) {
+    *location += (alignment - mod);
+  }
+}
+
+static uint8_t PeekByte(unsigned long location, CFDataRef data) {
+  uint8_t result;
+  CFRange range = CFRangeMake(location, 1);
+  CFDataGetBytes(data, range, &result);
+  return result;
+}
+
+static bool IsStandardType(uint8_t type) {
+  switch (type) {
+    case FlutterStandardFieldNil:
+    case FlutterStandardFieldTrue:
+    case FlutterStandardFieldFalse:
+    case FlutterStandardFieldInt32:
+    case FlutterStandardFieldInt64:
+    case FlutterStandardFieldIntHex:
+    case FlutterStandardFieldFloat64:
+    case FlutterStandardFieldString:
+    case FlutterStandardFieldUInt8Data:
+    case FlutterStandardFieldInt32Data:
+    case FlutterStandardFieldInt64Data:
+    case FlutterStandardFieldFloat64Data:
+    case FlutterStandardFieldList:
+    case FlutterStandardFieldMap:
+    case FlutterStandardFieldFloat32Data:
```

Comment:
My point is this might cause mistakes if a new enum is added and this code is not updated. If there is no compiler warning, maybe we can add a comment, or use a different way to determine valid enums.

Reply:
I moved the check to be right next to the enum declaration and added a comment that it needs to be updated if it changes.